text/0000-otlp-http-json.md

```diff
@@ -0,0 +1,146 @@
+# OTLP: HTTP+JSON Transport Extension for OTLP
+
+This is a proposal to add HTTP Transport extension supporting json serialization for
+[OTLP](0035-opentelemetry-protocol.md) (OpenTelemetry Protocol).
+
+## Table of Contents
+
+* [Motivation](#motivation)
+* [OTLP/HTTP+JSON Protocol Details](#otlphttpjson-protocol-details)
+  * [JSON Mapping](#json-mapping)
+  * [Request](#request)
+  * [Response](#response)
+    * [Success](#success)
+    * [Failures](#failures)
+    * [Throttling](#throttling)
+    * [All Other Responses](#all-other-responses)
+  * [Connection](#connection)
+  * [Parallel Connections](#parallel-connections)
+
+## Motivation
+
+Protobuf is a relatively big dependency, which some clients are not willing to
+take. For example, webjs, iOS/Android (in some scenarios, the size of the installation package is limited, do not want to introduce protobuf dependencies). Plain JSON is a smaller dependency and is built in the standard
+libraries of many programming languages.
+
+## OTLP/HTTP+JSON Protocol Details
+
+OTLP/HTTP+JSON will be consistent with the [OTLP/HTTP](0099-otlp-http.md) specification except that the payload will use JSON instead of protobuf.
+
+### JSON Mapping
+
+Use proto3 standard defined [JSON Mapping](https://developers.google.com/protocol-buffers/docs/proto3#json) for mapping between protobuf and json. `trace_id`  and `span_id` is base64 encoded in OTLP/HTTP+JSON, not hex.
+
+### Request
+
+Telemetry data is sent via HTTP POST request.
+
+The default URL path for requests that carry trace data is `/v1/traces` (for
+example the full URL when connecting to "example.com" server will be
+`https://example.com/v1/traces`). The request body is a JSON-encoded
+[`ExportTraceServiceRequest`](https://github.com/open-telemetry/opentelemetry-proto/blob/e6c3c4a74d57f870a0d781bada02cb2b2c497d14/opentelemetry/proto/collector/trace/v1/trace_service.proto#L38)
+message.
+
+The default URL path for requests that carry metric data is `/v1/metrics` and the
+request body is a JSON-encoded
+[`ExportMetricsServiceRequest`](https://github.com/open-telemetry/opentelemetry-proto/blob/e6c3c4a74d57f870a0d781bada02cb2b2c497d14/opentelemetry/proto/collector/metrics/v1/metrics_service.proto#L35)
+message.
```

Comment:
I assume this is copied verbatim from OTEP0099, except it says "JSON-encoded" instead of "ProtoBuf-encoded", right? (It is good to copy, I just want to confirm it is the only difference).

Reply:
Yes, this is the only difference.